install flutter_rust_bridge_codegen fail

### Describe the bug

install flutter_rust_bridge_codegen fail


os: mac m4 15.5 (24F74)
rust version: cargo 1.88.0 (873a06493 2025-05-10)


### Steps to reproduce

install flutter_rust_bridge_codegen


### Logs

```shell
➜  ~ cargo install flutter_rust_bridge_codegen
    Updating crates.io index
warning: spurious network error (3 tries remaining): [35] SSL connect error (Recv failure: Connection reset by peer)
  Downloaded flutter_rust_bridge_codegen v2.10.0
  Downloaded 1 crate (329.5KiB) in 6.47s
  Installing flutter_rust_bridge_codegen v2.10.0
    Updating crates.io index
     Locking 205 packages to latest compatible versions
      Adding cargo_metadata v0.14.2 (available: v0.20.0)
      Adding cargo_toml v0.21.0 (available: v0.22.1)
      Adding cbindgen v0.28.0 (available: v0.29.0)
      Adding convert_case v0.5.0 (available: v0.8.0)
      Adding enum-iterator v1.5.0 (available: v2.1.0)
      Adding fern v0.6.2 (available: v0.7.1)
      Adding indicatif v0.17.11 (available: v0.18.0)
      Adding itertools v0.10.5 (available: v0.14.0)
      Adding notify v6.1.1 (available: v8.1.0)
      Adding notify-debouncer-mini v0.4.1 (available: v0.6.0)
      Adding serial_test v2.0.0 (available: v3.2.0)
      Adding strum v0.26.3 (available: v0.27.1)
      Adding strum_macros v0.26.4 (available: v0.27.1)
      Adding toml v0.5.11 (available: v0.8.23)
  Downloaded addr2line v0.24.2
  Downloaded anyhow v1.0.98
  Downloaded adler2 v2.0.1
  Downloaded fsevent-sys v4.1.0
  Downloaded equivalent v1.0.2
  Downloaded futures-sink v0.3.31
  Downloaded futures-io v0.3.31
  Downloaded enum-iterator v1.5.0
  Downloaded is_terminal_polyfill v1.70.1
  Downloaded autocfg v1.5.0
  Downloaded anstyle-query v1.1.3
  Downloaded anstyle v1.0.11
  Downloaded cargo-platform v0.1.9
  Downloaded anstyle-parse v0.2.7
  Downloaded colorchoice v1.0.4
  Downloaded clap_lex v0.7.5
  Downloaded cargo_metadata v0.14.2
  Downloaded include_dir_macros v0.7.4
  Downloaded anstream v0.6.19
  Downloaded serial_test_derive v2.0.0
  Downloaded include_dir v0.7.4
  Downloaded sha1 v0.10.6
  Downloaded enum-iterator-derive v1.4.0
  Downloaded unit-prefix v0.5.1
  Downloaded number_prefix v0.4.0
  Downloaded indicatif-log-bridge v0.2.3
  Downloaded notify-debouncer-mini v0.4.1
  Downloaded strum v0.26.3
  Downloaded cfg-if v1.0.1
  Downloaded futures-core v0.3.31
  Downloaded errno v0.3.13
  Downloaded futures-executor v0.3.31
  Downloaded filetime v0.2.25
  Downloaded block-buffer v0.10.4
  Downloaded futures-task v0.3.31
  Downloaded fastrand v2.3.0
  Downloaded either v1.15.0
  Downloaded paste v1.0.15
  Downloaded is-terminal v0.4.16
  Downloaded glob v0.3.2
  Downloaded clap_derive v4.5.40
  Downloaded serial_test v2.0.0
  Downloaded heck v0.5.0
  Downloaded generic-array v0.14.7
  Downloaded crypto-common v0.1.6
  Downloaded cargo_toml v0.21.0
  Downloaded pin-utils v0.1.0
  Downloaded lazy_static v1.5.0
  Downloaded hex v0.4.3
  Downloaded heck v0.4.1
  Downloaded cpufeatures v0.2.17
  Downloaded version_check v0.9.5
  Downloaded quote v1.0.40
  Downloaded iana-time-zone v0.1.63
  Downloaded dashmap v5.5.3
  Downloaded core-foundation-sys v0.8.7
  Downloaded convert_case v0.5.0
  Downloaded bitflags v2.9.1
  Downloaded pathdiff v0.2.3
  Downloaded notify v6.1.1
  Downloaded itoa v1.0.15
  Downloaded futures-channel v0.3.31
  Downloaded enum_dispatch v0.3.13
  Downloaded digest v0.10.7
  Downloaded colored v1.9.4
  Downloaded serde_spanned v0.6.9
  Downloaded scopeguard v1.2.0
  Downloaded walkdir v2.5.0
  Downloaded smallvec v1.15.1
  Downloaded slab v0.4.10
  Downloaded rustversion v1.0.21
  Downloaded log v0.4.27
  Downloaded lock_api v0.4.13
  Downloaded getrandom v0.3.3
  Downloaded crossbeam-utils v0.8.21
  Downloaded clap v4.5.40
  Downloaded camino v1.1.10
  Downloaded topological-sort v0.2.2
  Downloaded toml_write v0.1.2
  Downloaded toml_datetime v0.6.11
  Downloaded strsim v0.11.1
  Downloaded rustc-demangle v0.1.25
  Downloaded proc-macro2 v1.0.95
  Downloaded pin-project-lite v0.2.16
  Downloaded parking_lot_core v0.9.11
  Downloaded once_cell v1.21.3
  Downloaded futures v0.3.31
  Downloaded console v0.16.0
  Downloaded console v0.15.11
  Downloaded same-file v1.0.6
  Downloaded semver v1.0.26
  Downloaded parking_lot v0.12.4
  Downloaded ryu v1.0.20
  Downloaded toml v0.8.23
  Downloaded itertools v0.10.5
  Downloaded hashbrown v0.15.4
  Downloaded clap_builder v4.5.40
  Downloaded futures-util v0.3.31
  Downloaded aho-corasick v1.1.3
  Downloaded hashbrown v0.14.5
  Downloaded indexmap v2.10.0
  Downloaded memchr v2.7.5
  Downloaded tempfile v3.20.0
  Downloaded crossbeam-channel v0.5.15
  Downloaded backtrace v0.3.75
  Downloaded portable-atomic v1.11.1
  Downloaded serde_derive v1.0.219
  Downloaded indicatif v0.17.11
  Downloaded derivative v2.2.0
  Downloaded num-traits v0.2.19
  Downloaded miniz_oxide v0.8.9
  Downloaded utf8parse v0.2.2
  Downloaded strum_macros v0.26.4
  Downloaded toml v0.5.11
  Downloaded cbindgen v0.28.0
  Downloaded serde v1.0.219
  Downloaded typenum v1.18.0
  Downloaded chrono v0.4.41
  Downloaded serde_json v1.0.140
  Downloaded gimli v0.31.1
  Downloaded indicatif v0.18.0
  Downloaded object v0.36.7
  Downloaded regex-syntax v0.8.5
  Downloaded fern v0.6.2
  Downloaded winnow v0.7.11
  Downloaded rustix v1.0.7
  Downloaded toml_edit v0.22.27
  Downloaded serde_yaml v0.9.34+deprecated
  Downloaded regex v1.11.1
  Downloaded unsafe-libyaml v0.2.11
  Downloaded unicode-ident v1.0.18
  Downloaded regex-automata v0.4.9
  Downloaded libc v0.2.174
  Downloaded syn v1.0.109
  Downloaded unicode-width v0.2.1
  Downloaded syn v2.0.104
  Downloaded 136 crates (8.9MiB) in 2.19s
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling libc v0.2.174
   Compiling serde v1.0.219
   Compiling cfg-if v1.0.1
   Compiling memchr v2.7.5
   Compiling autocfg v1.5.0
   Compiling once_cell v1.21.3
   Compiling version_check v0.9.5
   Compiling typenum v1.18.0
   Compiling equivalent v1.0.2
   Compiling log v0.4.27
   Compiling hashbrown v0.15.4
   Compiling portable-atomic v1.11.1
   Compiling generic-array v0.14.7
   Compiling parking_lot_core v0.9.11
   Compiling lock_api v0.4.13
   Compiling indexmap v2.10.0
   Compiling unicode-width v0.2.1
   Compiling futures-core v0.3.31
   Compiling futures-sink v0.3.31
   Compiling bitflags v2.9.1
   Compiling crossbeam-utils v0.8.21
   Compiling futures-channel v0.3.31
   Compiling num-traits v0.2.19
   Compiling pin-utils v0.1.0
   Compiling utf8parse v0.2.2
   Compiling pin-project-lite v0.2.16
   Compiling ryu v1.0.20
   Compiling futures-task v0.3.31
   Compiling winnow v0.7.11
   Compiling rustix v1.0.7
   Compiling toml_write v0.1.2
   Compiling slab v0.4.10
   Compiling object v0.36.7
   Compiling smallvec v1.15.1
   Compiling itoa v1.0.15
   Compiling futures-io v0.3.31
   Compiling scopeguard v1.2.0
   Compiling serde_json v1.0.140
   Compiling getrandom v0.3.3
   Compiling futures-util v0.3.31
   Compiling quote v1.0.40
   Compiling syn v2.0.104
   Compiling errno v0.3.13
   Compiling anstyle-parse v0.2.7
   Compiling core-foundation-sys v0.8.7
   Compiling semver v1.0.26
   Compiling adler2 v2.0.1
   Compiling rustversion v1.0.21
   Compiling camino v1.1.10
   Compiling anstyle v1.0.11
   Compiling is_terminal_polyfill v1.70.1
   Compiling syn v1.0.109
   Compiling anstyle-query v1.1.3
   Compiling lazy_static v1.5.0
   Compiling same-file v1.0.6
   Compiling heck v0.5.0
   Compiling gimli v0.31.1
   Compiling colorchoice v1.0.4
   Compiling anstream v0.6.19
   Compiling walkdir v2.5.0
   Compiling miniz_oxide v0.8.9
   Compiling serde_derive v1.0.219
   Compiling iana-time-zone v0.1.63
   Compiling crossbeam-channel v0.5.15
   Compiling addr2line v0.24.2
   Compiling futures-executor v0.3.31
   Compiling block-buffer v0.10.4
   Compiling crypto-common v0.1.6
   Compiling filetime v0.2.25
   Compiling fsevent-sys v4.1.0
   Compiling is-terminal v0.4.16
   Compiling console v0.16.0
   Compiling aho-corasick v1.1.3
   Compiling paste v1.0.15
   Compiling unit-prefix v0.5.1
   Compiling fastrand v2.3.0
   Compiling clap_lex v0.7.5
   Compiling cbindgen v0.28.0
   Compiling strsim v0.11.1
   Compiling anyhow v1.0.98
   Compiling rustc-demangle v0.1.25
   Compiling regex-syntax v0.8.5
   Compiling hashbrown v0.14.5
   Compiling dashmap v5.5.3
   Compiling backtrace v0.3.75
   Compiling clap_builder v4.5.40
   Compiling tempfile v3.20.0
   Compiling indicatif v0.18.0
   Compiling regex-automata v0.4.9
   Compiling colored v1.9.4
   Compiling notify v6.1.1
   Compiling digest v0.10.7
   Compiling chrono v0.4.41
   Compiling toml_datetime v0.6.11
   Compiling serde_spanned v0.6.9
   Compiling toml_edit v0.22.27
   Compiling cargo-platform v0.1.9
   Compiling toml v0.8.23
   Compiling futures v0.3.31
   Compiling serial_test_derive v2.0.0
   Compiling clap_derive v4.5.40
   Compiling enum-iterator-derive v1.4.0
   Compiling parking_lot v0.12.4
   Compiling include_dir_macros v0.7.4
   Compiling console v0.15.11
   Compiling cpufeatures v0.2.17
   Compiling either v1.15.0
   Compiling heck v0.4.1
   Compiling unsafe-libyaml v0.2.11
   Compiling number_prefix v0.4.0
   Compiling indicatif v0.17.11
   Compiling clap v4.5.40
   Compiling enum_dispatch v0.3.13
   Compiling itertools v0.10.5
   Compiling serde_yaml v0.9.34+deprecated
   Compiling sha1 v0.10.6
   Compiling include_dir v0.7.4
   Compiling enum-iterator v1.5.0
   Compiling serial_test v2.0.0
   Compiling cargo_toml v0.21.0
   Compiling cargo_metadata v0.14.2
   Compiling derivative v2.2.0
   Compiling regex v1.11.1
   Compiling fern v0.6.2
   Compiling toml v0.5.11
   Compiling notify-debouncer-mini v0.4.1
   Compiling strum_macros v0.26.4
   Compiling indicatif-log-bridge v0.2.3
   Compiling glob v0.3.2
   Compiling strum v0.26.3
   Compiling convert_case v0.5.0
   Compiling hex v0.4.3
   Compiling topological-sort v0.2.2
   Compiling pathdiff v0.2.3
   Compiling flutter_rust_bridge_codegen v2.10.0
error[E0308]: mismatched types
  --> /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/flutter_rust_bridge_codegen-2.10.0/src/library/utils/logs.rs:46:61
   |
46 |     let log_wrapper = indicatif_log_bridge::LogWrapper::new(MULTI_PROGRESS.clone(), fern_logger);
   |                       ------------------------------------- ^^^^^^^^^^^^^^^^^^^^^^ expected `indicatif::multi::MultiProgress`, found `MultiProgress`
   |                       |
   |                       arguments to this function are incorrect
   |
note: two different versions of crate `indicatif` are being used; two types coming from two different versions of the same crate are different types even if they look the same
   |
  ::: /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/indicatif-0.18.0/src/multi.rs:18:1
   |
18 | pub struct MultiProgress {
   | ^^^^^^^^^^^^^^^^^^^^^^^^ this is the expected type `indicatif::multi::MultiProgress`
  --> /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/indicatif-0.17.11/src/multi.rs:18:1
   |
18 | pub struct MultiProgress {
   | ^^^^^^^^^^^^^^^^^^^^^^^^ this is the found type `MultiProgress`
   |
  ::: /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/flutter_rust_bridge_codegen-2.10.0/src/library/utils/logs.rs:46:23
   |
46 |     let log_wrapper = indicatif_log_bridge::LogWrapper::new(MULTI_PROGRESS.clone(), fern_logger);
   |                       -------------------- one version of crate `indicatif` used here, as a dependency of crate `indicatif_log_bridge`
   |
  ::: /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/flutter_rust_bridge_codegen-2.10.0/src/library/utils/console.rs:1:5
   |
1  | use indicatif::{MultiProgress, ProgressBar, ProgressState, ProgressStyle};
   |     --------- one version of crate `indicatif` used here, as a direct dependency of the current crate
   = help: you can use `cargo tree` to explore your dependency tree
note: associated function defined here
  --> /Users/m4/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/indicatif-log-bridge-0.2.3/src/lib.rs:67:12
   |
67 |     pub fn new(bar: MultiProgress, log: L) -> Self {
   |            ^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `flutter_rust_bridge_codegen` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `flutter_rust_bridge_codegen v2.10.0`, intermediate artifacts can be found at `/var/folders/cj/y592z0t90yq391w26r9sn58r0000gn/T/cargo-installer9X81`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

### Expected behavior

_No response_

### Generated binding code

```shell

```

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

```shell

```

### Version of `clang++`

_No response_

### Additional context

_No response_

--- a/frb_codegen/Cargo.toml
+++ b/frb_codegen/Cargo.toml
@@ -48,7 +48,7 @@ derivative = "2.2.0"
 paste = "1.0.14"
 proc-macro2 = { version = "1.0.66", features = ["span-locations"] }
 include_dir = "0.7.3"
-indicatif = "0.17.7"
+indicatif = ">=0.17.7, <0.18.0"
 indicatif-log-bridge = "0.2.2"
 notify = "6.1.1"
 notify-debouncer-mini = "0.4.1"

